packages/react-components/react-tree/src/utils/createFlatTreeItems.ts

```diff
@@ -0,0 +1,123 @@
+import type { TreeItemId } from '../TreeItem';
+import type { ImmutableSet } from './ImmutableSet';
+import type { FlatTreeItem, FlatTreeItemProps, MutableFlatTreeItem } from '../hooks/useFlatTree';
+
+/**
+ * @internal
+ */
+export type FlatTreeItems = {
+  size: number;
+  root: FlatTreeItem;
+  get(id: string): FlatTreeItem | undefined;
+  set(id: string, value: FlatTreeItem): void;
+  getByIndex(index: number): FlatTreeItem;
+};
+
+/**
+ * creates a list of flat tree items
+ * and provides a map to access each item by id
+ */
+export function createFlatTreeItems(flatTreeItemProps: FlatTreeItemProps[]): FlatTreeItems {
+  const root = createFlatTreeRootItem();
+  const itemsPerId = new Map<string, MutableFlatTreeItem>([[flatTreeRootId, root]]);
+  const items: MutableFlatTreeItem[] = [];
+
+  for (let index = 0; index < flatTreeItemProps.length; index++) {
+    const { parentId = flatTreeRootId, ...treeItemProps } = flatTreeItemProps[index];
+
+    const nextItemProps: FlatTreeItemProps | undefined = flatTreeItemProps[index + 1];
+    const currentParent = itemsPerId.get(parentId);
+    if (!currentParent) {
+      if (process.env.NODE_ENV === 'development') {
+        // eslint-disable-next-line no-console
+        console.error(
+          `useFlatTree: item ${flatTreeItemProps[index].id} not properly initialized, make sure provided items are organized`,
```

Comment:
nit: let's make this error message clearer, 'not properly initializes' is not very clear on what I did wrong if I got this error message.